browser/src/app/DocumentView.ts

```diff
@@ -45,6 +45,11 @@ class DocumentViewBase {
 		const groups: any = [];
 		groups.length = 0;
 
+		for (let i = rectangles.length - 1; i > -1; i--) {
+			if (rectangles[i][3] === 0 || rectangles[i][2] === 0)
+				rectangles.splice(i, 1); // Remove the 0-width or 0-height rectangle from the list.
+		}
```

Comment:
tip: you could use `Array.filter` -> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

for clean and short code :)

`rectangles = rectangles.filter((rectangle) => { return rectangle[i][3] !== 0 || rectangle[i][2] !== 0; });`